src/plugins/lua/bindings/lua_ImGuiCustom.cpp

```diff
@@ -99,6 +102,54 @@ void RegisterBindings_ImGuiCustom(sol::table& ImGui)
 			[](mq::imgui::ConsoleWidget* pThis, const ImVec4& col, std::string_view text) { pThis->AppendText(text, MQColor(col), false); }
 		)
 	);
-}
 
+	// Widgets: ZepInputBox
+	ImGui.new_usertype<mq::imgui::ImGuiZepEditor>(
+	"ZepEditor", sol::factories(&mq::imgui::ImGuiZepEditor::Create),
+		"Render", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis) { pThis->Render(); },
+			[](mq::imgui::ImGuiZepEditor* pThis, ImVec2 displaySize) { pThis->Render(displaySize); }
+		),
+		"Clear", &mq::imgui::ImGuiZepEditor::Clear,
+		"SetSyntax", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis, const char* syntaxName) { pThis->SetSyntaxProvider(syntaxName); }
+		),
+		"SetText", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis, std::string_view text) { pThis->SetText(text, false); }
+		),
+		"GetText", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis) -> std::string { size_t bufferSize = pThis->GetTextLength(); std::string text; text.reserve(bufferSize); pThis->GetText(text); return text; }
+		),
+		"SetFontSize", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis, int size) { pThis->SetFontSize(size); }
+		),
+		"GetFontSize", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis) -> int { return pThis->GetFontSize(); }
+		),
+		"SetWindowFlags", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis, int flags) { pThis->SetWindowFlags(flags); }
+		),
+		"GetWindowFlags", sol::overload(
+			[](mq::imgui::ImGuiZepEditor* pThis) -> int { return pThis->GetWindowFlags(); }
+		)
+	);
```

Comment:
@DerpleMQ2 These `sol::overloads` are unnecessary. `sol::overload` is only needed if you have more than one function with the same name (hence "overload")

Reply:
I don't know if it's possible, but it would be pretty nice if you could just edit the string buffer that the editor is displaying and that calls SetText and Render for you

Reply:
Addressed the naming and interface complexity with my latest commit

Reply:
> @DerpleMQ2 These `sol::overloads` are unnecessary. `sol::overload` is only needed if you have more than one function with the same name (hence "overload")

Fixed

Reply:
Simplified